### PR TITLE
Fix/mark pools as existing

### PR DESCRIPTION
**Pull request by mwamedacen**
[/] UniswapV3
[/] PancakeSwapV3 (fork of UniswapV3)
[/] Algebra

resolves BACK-1274

### CODE REVIEW

**Comment by JackieJoo** [comment]
## UniswapV3

### Before Pool Creation - we have a pair in `*not_existings_pool_set`

Screenshot 2023-10-02 at 18 06 24


### Pool Creation

https://arbiscan.io/tx/0x81b49a1e84999cb8c9ff3d66ad0e0a96220d98499762515dd704969e5f3dde04

### After Pool Creation - we don't have a pair in `*not_existings_pool_set`

Screenshot 2023-10-02 at 18 07 16

## PancakeV3

### Before Pool Creation - we have a pair in `*not_existings_pool_set`

Screenshot 2023-10-02 at 18 07 59


### Pool Creation

https://arbiscan.io/tx/0x5376b96b8d1b2bcefddff8ef60e95ef29513baf4dddc5fdafe669e9019b8506c

### After Pool Creation - we don't have a pair in `*not_existings_pool_set`

Screenshot 2023-10-02 at 18 11 18


## Algebra (CamelotV3)

### Before Pool Creation - we have a pair in `*not_existings_pool_set`

Screenshot 2023-10-02 at 18 12 36


### Pool Creation
https://arbiscan.io/tx/0x39bed538ccba7b239e3bba6bfe35e4708c321be460dba2f4aaa22dde5e891bf2

### After Pool Creation - we don't have a pair in `*not_existings_pool_set`

Screenshot 2023-10-02 at 18 13 31

File: src/dex/algebra/algebra-factory.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    super(parentName, `${parentName} Factory`, dexHelper, logger, true, mapKey);
+
+    this.addressesSubscribed = [factoryAddress];
+
+    this.logDecoder = (log: Log) => this.factoryIface.parseLog(log);
+
+    this.handlers['Pool'] = this.handleNewPool.bind(this);
+  }
+
+  generateState(): FactoryState {
```
Comment (Verisana) [inline src/dex/algebra/algebra-factory.ts:48]: Why not an empty object as a placeholder state?

File: src/dex/algebra/algebra.ts [hunk 2]
```diff
@@ -147,6 +157,35 @@ export class Algebra extends SimpleExchange implements IDex<AlgebraData> {
...
+    // consider doing it only from master pool for less calls to distant cache
+
+    // delete entry locally to let local instance discover the pool
+    delete this.eventPools[this.getPoolIdentifier(_token0, _token1)];
+
+    this.logger.info(
+      `${logPrefix} delete pool from not existing set: ${poolKey}`,
+    );
+    // delete pool record from set, not waiting for result
+    this.dexHelper.cache.zrem(this.notExistingPoolSetKey, [poolKey]);
```
Comment (Verisana) [inline src/dex/algebra/algebra.ts:181]: I feel like better to add `catch`, so if it fails, it won't fail in arbitrary place, but will be contained

File: src/dex/algebra/algebra-factory.ts (hunk 1 above)
Comment (JackieJoo) [reply src/dex/algebra/algebra-factory.ts:48]: done

**Review by paraschiv-razvan — APPROVED** [review]

File: src/dex/algebra/algebra.ts [hunk 3]
```diff
@@ -114,6 +116,14 @@ export class Algebra extends SimpleExchange implements IDex<AlgebraData> {
 
     this.AlgebraPoolImplem =
       config.version === 'v1.1' ? AlgebraEventPoolV1_1 : AlgebraEventPoolV1_9;
+
+    this.factory = new AlgebraFactory(
+      dexHelper,
+      dexKey,
+      this.config.factory,
+      this.logger,
+      this.onPoolCreatedDeleteFromNonExistingSet.bind(this),
```
Comment (mwamedacen) [inline src/dex/algebra/algebra.ts:125]: we don't need to rebind if we use arrow functions

**Review by aburkut — APPROVED** [review]

File: src/dex/algebra/algebra.ts (hunk 3 above)
Comment (JackieJoo) [reply src/dex/algebra/algebra.ts]: that's right, fixed

**Pull request merged**